### PR TITLE
test: Add tests for Retry-After Header

### PR DESCRIPTION
**Pull request by philipphofmann**
Add tests for regular rate limit response with 429
status code and a Retry-After Header for SentryHttpTransport.
This is just a first set of tests. More tests should follow.

Please pay special attention to https://github.com/getsentry/sentry-cocoa/blob/da24ed1874979768a5e057d336f051e2090c48c0/Sources/Sentry/SentryHttpTransport.m#L266 
[HttpDate is always expressed in GMT](ttps://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date). I hope this doesn't break anything.

### CODE REVIEW

File: Sources/Sentry/SentryHttpTransport.m [hunk 1]
```diff
@@ -259,6 +260,11 @@ - (NSDate *)parseRetryAfterHeader:(NSString * __nullable)retryAfterHeader {
     // parsing as double/seconds failed, try to parse as date
     NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
```
Comment (HazAT) [inline Sources/Sentry/SentryHttpTransport.m:261]: I know this wasn't introduced in this PR and but before we go GA
I don't think we need to create a dateformatter for every call in `parseRetryAfterHeader`.
Initializing one is expensive and maybe we should only do it once per transport.

Comment (bruno-garcia) [reply Sources/Sentry/SentryHttpTransport.m:261]: > NSDateFormatter is not thread safe

https://developer.apple.com/documentation/foundation/nsdateformatter

Sounds like we either need to create one per request or we'd need to create a pool

Comment (bruno-garcia) [reply Sources/Sentry/SentryHttpTransport.m:261]: Also worth noting that *today* Sentry returns seconds, not date.

Comment (bruno-garcia) [reply Sources/Sentry/SentryHttpTransport.m:261]: I read it again now and I skimmed over important bits:

> In macOS 10.9 and later NSDateFormatter is thread safe so long as you are using the modern behavior in a 64-bit app.
On earlier versions of the operating system, or when using the legacy formatter behavior or running in 32-bit in macOS, NSDateFormatter is not thread safe

Comment (philipphofmann) [reply Sources/Sentry/SentryHttpTransport.m:261]: I am going to address this issue in another PR. I am already on it.

Comment (philipphofmann) [reply Sources/Sentry/SentryHttpTransport.m:261]: I opened another PR for this https://github.com/getsentry/sentry-cocoa/pull/440

**Pull request merged**